src/wrappers/generic/GenericExecutor.ts

```diff
@@ -0,0 +1,79 @@
+import Utils from '../../utils/Utils';
+import {
+    ClassNames,
+    DomainNames,
+    GenericWrapperTags,
+    SpanTags,
+    SpanTypes,
+} from '../../Constants';
+import PluginContext from '../../plugins/PluginContext';
+import ExecutionContext from '../../context/ExecutionContext';
+import WrapperUtils from '../WebWrapperUtils';
+import InvocationSupport from '../../plugins/support/InvocationSupport';
+import GooglePubSubUtils from '../../utils/GooglePubSubUtils';
+
+const get = require('lodash.get');
+
+/**
+ * Start invocation process
+ * @param {PluginContext} pluginContext
+ * @param {ExecutionContext} execContext
+ */
+export function startInvocation(pluginContext: PluginContext, execContext: ExecutionContext) {
+    execContext.invocationData = WrapperUtils.createInvocationData(execContext, pluginContext);
+}
+
+/**
+ * Finish invocation process
+ * @param {PluginContext} pluginContext
+ * @param {ExecutionContext} execContext
+ */
+export function finishInvocation(pluginContext: PluginContext, execContext: ExecutionContext) {
+    WrapperUtils.finishInvocationData(execContext, pluginContext);
+}
+
+/**
+ * Start trace
+ * @param {PluginContext} pluginContext
+ * @param {ExecutionContext} execContext
+ */
+export function startTrace(pluginContext: PluginContext, execContext: ExecutionContext) {
+
+    const {
+        request,
+        tracer,
+    } = execContext;
+
+    const functionName = get(request, `functionName`);
```

Comment:
Can function name be empty? for ex. anonymous functions?

Reply:
Yes, it can.  What should our behavior be in this situation ? maybe, we can give default value like thundra-function ?

Reply:
On current behavior, tags value will be empy string. ( "" )

Reply:
Empty tag value is not a problem but the problem is empty operation name as we pass the function name as operation name below